### PR TITLE
Fix cache expiration check

### PR DESCRIPTION
**Pull request by liggitt**
The check for whether an entry in the `forceLiveLookup` cache had expired was backwards. Fixed the logic and added tests

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33464)

### CODE REVIEW

**Comment by liggitt** [comment]
@derekwaynecarr @deads2k

**Comment by deads2k** [comment]
> The check for whether an entry in the forceLiveLookup cache had expired was backwards. Fixed the logic and added tests

There's a type in kube to do this.

**Comment by deads2k** [comment]
`LRUExpireCache` I think

**Comment by liggitt** [comment]
updated to use LRUExpireCache

File: pkg/util/cache/lruexpirecache.go [hunk 1]
```diff
@@ -23,13 +23,32 @@ import (
 	"github.com/golang/groupcache/lru"
 )
 
+// Clock defines an interface for obtaining the current time
+type Clock interface {
+	Now() time.Time
+}
+
+// realClock implements the Clock interface by calling time.Now()
+type realClock struct{}
```
Comment (deads2k) [inline pkg/util/cache/lruexpirecache.go]: Just to avoid depending on `pkg/util/clock`?  Why bother?

**Comment by deads2k** [comment]
I would have let one util package depend on another, but this is fine.

We have other uses in limit ranger and resourcequota.  Want to update here or separately?

lgtm

**Comment by liggitt** [comment]
> We have other uses in limit ranger and resourcequota. Want to update here or separately?

separately, those were handling missing/expired cases correctly already

**Comment by derekwaynecarr** [comment]
lgtm

**Comment by liggitt** [comment]
unit test failed with #32455
@k8s-bot unit test this

**Pull request merged**